compiler/testData/diagnostics/tests/smartCasts/publicVals/open.fir.kt

```diff
@@ -7,5 +7,5 @@ infix fun Int.bar(i: Int) = i
 fun test() {
     val p = A()
     // For open value properties, smart casts should not work
-    if (p.foo is Int) p.foo <!UNSAFE_INFIX_CALL!>bar<!> 11
+    if (p.foo is Int) <!UNSAFE_CALL!>p.foo bar 11<!>
```

Comment:
Same here. But... I'm suddenly confused how your change can affect this...

Reply:
Also reverted here. The reason is because the previous logic in org.jetbrains.kotlin.fir.analysis.diagnostics.ConeDiagnosticToFirDiagnosticKt#mapUnsafeCallError relies on source being the `OPERATION_REFERENCE`. I have updatd that logic to check for an immediate child instead.